src/apps/api/views/leaderboards.py

```diff
@@ -43,15 +44,24 @@ def update(self, request, *args, **kwargs):
         return response
 
 
+def is_user_submission_owner_or_admin(request, submission, competition):
+    if request.user.is_superuser or request.user in competition.collaborators.all():
+        return True
+    elif request.user == submission.owner:
+        return True
+    else:
+        raise drf_exceptions.AuthenticationFailed("You do not have permission to change this submission's leaderboard status")
+
+
 @api_view(['POST'])
 @permission_classes((IsAuthenticated, ))
 def add_submission_to_leaderboard(request, submission_pk):
```

Comment:
Make this an action on the `SubmissionViewSet`.

Reply:
Done